CRM/Core/BAO/File.php

```diff
@@ -40,6 +40,27 @@ class CRM_Core_BAO_File extends CRM_Core_DAO_File {
 
   static $_signableFields = array('entityTable', 'entityID', 'fileID');
 
+  /**
+   * Takes an associative array and creates a File object.
+   *
+   * @param array $params
+   *   (reference ) an assoc array of name/value pairs.
+   *
+   * @return CRM_Core_BAO_File
+   */
+  public static function create($params) {
+    $fileDAO = new CRM_Core_DAO_File();
+    $fileDAO->copyValues($params);
+
+    if (empty($params['id']) && empty($params['created_id'])) {
+      $fileDAO->created_id = CRM_Core_Session::getLoggedInContactID();
+    }
+
+    $fileDAO->save();
```

Comment:
I think this BAO::create() function needs to call pre/post hooks.

Reply:
oops missed it, thanks for pointing that out.

 Added the hook calls. Can you please check now?